First lab: up to step 3

# Step 1

In [19]:
import pandas as pd
df = pd.read_csv("Dataset1.csv")
print("(observations, variables) = ", df.shape)

missing = df.isna().sum()
print("Missing values in each column:\n", missing)

class_balance = df["Purchased"].value_counts()
print("Class balance:\n", class_balance)

(observations, variables) =  (400, 5)
Missing values in each column:
 User ID            0
Gender             0
Age                0
EstimatedSalary    0
Purchased          0
dtype: int64
Class balance:
 Purchased
0    257
1    143
Name: count, dtype: int64


Questions:
- 400 observations, 5 variables (user id, gender, age, estimated salary, purchased)
- The target has for Purchased, 257 in class 0 and 143 in class 1. This is not balanced. 
- No missing values in any column

In [20]:
predictors = df.drop(columns=["Purchased"])
target = df["Purchased"]

# Step 2

# Step 3

# Step 4

# Step 5

# Step 6

# Step 7

# Step 8

# Step 9